src/index.js

```diff
@@ -246,7 +256,7 @@ export function makeUseAxios(configureOptions) {
         request(withCancelToken(config), options, dispatch).catch(() => {})
       }
 
-      return cancelOutstandingRequest
+      return () => cancelOutstandingRequest(options.manual)
```

Comment:
@simoneb this handles the goal of #372 . When `manual: true`, the config object itself could be changed to cancel the request.

If this isn't needed, we could remove the parameter on `cancelOutstandingRequest`
and update `cancelManually`:
```js
const cancelManually = React.useCallback(() => {
  cancelOutstandingRequest()
  dispatch({ type: actions.REQUEST_CANCELLED })
}
```
With this, the changes on lines 228 - 234, and 259 could be removed. This would mean that we only reset `loading` when using the manualCancel return callback.